include/particles/particle_finite_strain.tcc

```diff
@@ -259,6 +259,161 @@ void mpm::ParticleFiniteStrain<Tdim>::initialise_constitutive_law() noexcept {
           &state_variables_[mpm::ParticlePhase::Solid]);
 }
 
+//! Map mass and material stiffness matrix to cell
+//! (used in poisson equation LHS)
+template <unsigned Tdim>
+inline bool mpm::ParticleFiniteStrain<Tdim>::map_stiffness_matrix_to_cell(
+    double newmark_beta, double dt, bool quasi_static) {
+  bool status = true;
+  try {
+    // Check if material ptr is valid
+    assert(this->material() != nullptr);
+
+    // Compute material stiffness matrix
+    this->map_material_stiffness_matrix_to_cell();
+
+    // Compute mass matrix
+    if (!quasi_static) this->map_mass_matrix_to_cell(newmark_beta, dt);
+
+    // Compute geometric stiffness matrix
+    this->map_geometric_stiffness_matrix_to_cell();
+
+  } catch (std::exception& exception) {
+    console_->error("{} #{}: {}\n", __FILE__, __LINE__, exception.what());
+    status = false;
+  }
+  return status;
+}
+
+//! Map geometric stiffness matrix to cell (used in equilibrium equation LHS)
+template <unsigned Tdim>
+inline bool
+    mpm::ParticleFiniteStrain<Tdim>::map_geometric_stiffness_matrix_to_cell() {
+  bool status = true;
+  try {
+    // Calculate G matrix
+    const Eigen::MatrixXd gmatrix = this->compute_gmatrix();
+
+    // Stress component matrix
+    Eigen::MatrixXd stress_matrix = compute_stress_matrix();
```

Comment:
Let's discuss this today. I think there is a much shorter and less complicated approach.